core/src/test/java/org/apache/calcite/test/SqlFunctionsTest.java

```diff
@@ -810,6 +826,674 @@ private void thereAndBack(byte[] bytes) {
+  }
+
+  @Test public void testJsonExists() {
+    assertThat(
+        SqlFunctions.
+            jsonExists(SqlFunctions.PathContext
+                .withReturned(SqlFunctions.PathMode.STRICT,
+                "bar"), SqlJsonExistsErrorBehavior.FALSE),
+        is(true));
+    assertThat(
+        SqlFunctions.
+            jsonExists(SqlFunctions.PathContext
+                .withReturned(SqlFunctions.PathMode.STRICT,
+                "bar"), SqlJsonExistsErrorBehavior.TRUE),
+        is(true));
+    assertThat(
+        SqlFunctions.
+            jsonExists(SqlFunctions.PathContext
+                .withReturned(SqlFunctions.PathMode.STRICT,
+                "bar"), SqlJsonExistsErrorBehavior.UNKNOWN),
+        is(true));
+    assertThat(
+        SqlFunctions.
+            jsonExists(SqlFunctions.PathContext
+                .withReturned(SqlFunctions.PathMode.STRICT,
+                "bar"), SqlJsonExistsErrorBehavior.ERROR),
+        is(true));
+    assertThat(
+        SqlFunctions.
+            jsonExists(SqlFunctions.PathContext
+                .withReturned(SqlFunctions.PathMode.LAX,
+                null), SqlJsonExistsErrorBehavior.FALSE),
+        is(false));
+    assertThat(
+        SqlFunctions.
+            jsonExists(SqlFunctions.PathContext
+                .withReturned(SqlFunctions.PathMode.LAX,
+                null), SqlJsonExistsErrorBehavior.TRUE),
+        is(false));
+    assertThat(
+        SqlFunctions.
+            jsonExists(SqlFunctions.PathContext
+                .withReturned(SqlFunctions.PathMode.LAX,
+                null), SqlJsonExistsErrorBehavior.UNKNOWN),
+        is(false));
+    assertThat(
+        SqlFunctions.
+            jsonExists(SqlFunctions.PathContext
+                .withReturned(SqlFunctions.PathMode.LAX,
+                null), SqlJsonExistsErrorBehavior.ERROR),
+        is(false));
+    assertThat(
+        SqlFunctions.
+            jsonExists(SqlFunctions.PathContext
+                    .withStrictException(new Exception("test message")),
+                SqlJsonExistsErrorBehavior.FALSE),
+        is(false));
+    assertThat(
+        SqlFunctions.
+            jsonExists(SqlFunctions.PathContext
+                    .withStrictException(new Exception("test message")),
+                SqlJsonExistsErrorBehavior.TRUE),
+        is(true));
+    assertThat(
+        SqlFunctions.
+            jsonExists(SqlFunctions.PathContext
+                    .withStrictException(new Exception("test message")),
+                SqlJsonExistsErrorBehavior.UNKNOWN),
+        nullValue());
+
+    try {
+      SqlFunctions.
+          jsonExists(SqlFunctions.PathContext
+                  .withStrictException(new Exception("test message")),
+              SqlJsonExistsErrorBehavior.ERROR);
+      fail("expect exception, but not");
+    } catch (RuntimeException e) {
+      assertThat(e.getMessage(), is("java.lang.Exception: test message"));
+    }
+  }
+
+  @Test public void testJsonValueAny() {
```

Comment:
Would you please ensure the tests print human-understandable messages in case of failures?